src/Home/CreateRoomButton.js

```diff
@@ -1,18 +1,42 @@
+import { useEffect } from "react";
 import { supabase } from "../supabase_client";
 import { generateRoomCode } from "../utils";
 
 const CreateRoomButton = (props) => {
-  const { player, setPage, setRoomCode, setIsHost } = props;
+  const { player, setPage, setRoomCode, setIsHost, roomCode } = props;
+  let pageSelect = "home";
+  let hostSelect = false;
+  let roomSelect = null;
```

Comment:
I think you want to use these as local variables in a function rather than in the component